src/ekat/kokkos/ekat_kokkos_types.hpp

```diff
@@ -71,4 +71,13 @@ struct KokkosTypes
 
 } // namespace ekat
 
+// Kokkos-compatible reduction identity for arbitrary packs
+namespace Kokkos {
+template<typename PackType>
+struct reduction_identity {
```

Comment:
I believe this should be a specialization, while this one is a declaration of a generic template. I'm not sure why it does not conflict with the one in `Kokkos_NumericTraits.hpp`.